synapse/config/workers.py

```diff
@@ -296,6 +306,43 @@ def read_config(self, config, **kwargs):
             self.worker_name is None and background_tasks_instance == "master"
         ) or self.worker_name == background_tasks_instance
```

Comment:
this should probably be fixed to use `instance_name` while you're here:

```suggestion
            self.instance_name == background_tasks_instance
        )
```